@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -502,20 +495,31 @@ function placePointLabelChoosingAnchor(
 
         // Glyphs arrangement have been changed remove text buffer object and apply
         // new text layout to the label.
-        if (i !== 0) {
+        if (!recentPlacementTest) {
             label.textBufferObject = undefined;
         }
+
+        if (!basePlacementTest) {
+            // Text elements with alternative placement require their own unique layout style.
+            // Provide private copy of label layout and attach it to current label's text canvas.
+            textCanvas.textLayoutStyle = label.layoutStyle!.clone();
+            // Setup TextCanvas layout settings of the new placement as it is required for further
+            // TextBufferObject allocations and measurements in addText().
+            textCanvas.textLayoutStyle.horizontalAlignment = anchorPlacement.h;
+            textCanvas.textLayoutStyle.verticalAlignment = anchorPlacement.v;
```

Comment:
placePointLabelAtAnchor already sets these alignments and then reverts them back. Just leave the aligments set already in placePointLabelAtAnchor.

Reply:
We already have that in previous version... but this way I need to revert them back every time the placement will fail (with rejected or invisible placement result). Keep in mind that rejected labels need to be rendered with the latest alignment that was successful (from previous frame) in order to fade out in the same position.
I have done it in new way just to simplify the logic -> override TextCanvas settings only if new & alternative placement have been found.